src/backend/binaryen/lib/interface/meta.c

```diff
@@ -9,65 +9,69 @@
 
 
 /**
- * MetaField record the properties information of object
+ * MetaProperty record the properties information of object
  * name: property name
  * flag_and_index: flag and index of the property
  * type: type of the property, represent by property's type id
 */
-typedef struct MetaField {
+typedef struct MetaProperty {
     char *name;
     int flag_and_index;
     int type;
-} MetaField;
+} MetaProperty;
 
 /**
  * Meta record type id and the properties information of object
  * type_id: type id of the object
- * count: number of fields
- * fields: fields information
+ * count: number of properties
+ * properties: properties information
  */
-
 typedef struct Meta {
     int type_id;
     int impl_id;
     int count;
-    MetaField fields[0];
+    MetaProperty properties[0];
 } Meta;
 
 #define META_FLAG_MASK 0x0000000F
 #define META_INDEX_MASK 0xFFFFFFF0
 
-/* find field index based on prop_name*/
-int find_index(Meta *meta, char *prop_name, int flag) {
-    MetaField f;
-    int f_flag;
-    int f_index;
-    int s_flag = flag & META_FLAG_MASK;
+/* find property index based on prop_name*/
+int find_property_flag_and_index(Meta *meta, char *prop_name, int flag) {
+    MetaProperty prop;
+    int target_flag = flag & META_FLAG_MASK;
+    int unknown_flag = 4 & META_FLAG_MASK;
```

Comment:
Don't use raw number here, use enum instead

Reply:
Get it.